kong/core/handler.lua

```diff
@@ -60,6 +60,10 @@ return {
         ngx.log(ngx.ERR, "failed the initial dns/balancer resolve: ", err)
         return responses.send_HTTP_INTERNAL_SERVER_ERROR()
       end
+
+      -- Websocket
+      local is_upgrade = ngx.var.http_connection and ngx.var.http_connection:lower() == "upgrade"
+      ngx.var.upstream_connection = is_upgrade and "upgrade" or "keep-alive"
```

Comment:
The logic changed here: it is relying on the value of the `Connection` header, and not `Upgrade` anymore. It should rely on `Upgrade`.

Also, any particular reason for not implementing this in a map anymore? 

Following those 2 rules, this could/should be:
```nginx
map $http_upgrade $upstream_connection {
    default ''; # assumed keepalive connection by default since this is HTTP/1.1
    websocket upgrade; # set 'Connection: Upgrade' if 'Upgrade: WebSocket'
}
```

I think the upstream `Upgrade` header should also receive particular attention, being cleared by default, and only forwarded when its value is `Upgrade: WebSocket`
```nginx
map $http_upgrade $upstream_upgrade {
    default '';
    websocket websocket;
}
```